core/src/commonMain/kotlin/com/xebia/functional/xef/vectorstores/LocalVectorStore.kt

```diff
@@ -31,15 +31,27 @@ private constructor(private val embeddings: Embeddings, private val state: Atomi
 
   override suspend fun addMemories(memories: List<Memory>) {
```

Comment:
A shorter option, but I don't know if less readable:
```kotlin
override suspend fun addMemories(memories: List<Memory>) {
    state.update { prevState ->
      prevState.copy(
        orderedMemories = memories.groupBy { it.conversationId }.let { memories ->
          (prevState.orderedMemories.keys + memories.keys).associateWith { key ->
            val l1 = prevState.orderedMemories[key] ?: emptyList()
            val l2 = memories[key] ?: emptyList()
            l1 + l2
          }
        }
      )
    }
  }
```

Reply:
My opinion is: This is great!